### ISSUE TITLE
Y011: Allow `math.inf`, `math.e`, `math.pi`, and `math.tau` as simple defaults

### ISSUE THREAD

**Issue by XuehaiPan** [PyCQA/flake8-pyi#354]
Some math constants such as `float('inf')`, `math.inf`, `math.e`, `math.pi`, `math.tau`, and their negatives should also be allowed in `.pyi` as simple defaults.

For example:

```python
# test.pyi
import math

def norm1(vec: list[float], p: float = math.inf) -> float: ...
def norm2(vec: list[float], p: float = float('inf')) -> float: ...
```

Current output:

```console
$ flake8 --select=Y test.pyi
test.pyi:4:40: Y011 Only simple default values allowed for typed arguments
test.pyi:5:40: Y011 Only simple default values allowed for typed arguments
```

**Comment by AlexWaygood** [comment]
- `math.e`, `math.pi` and `math.tau` all seem very reasonable to me, as they're commonly used constants. I'd happily accept a PR so that we allow those.
- I wouldn't want to allow `float("inf")`, as we currently don't allow any "call expressions" as default values, and it's nice to be consistent about that
- `math.inf` I could go either way on. `math.inf == math.inf` apparently, so I guess it counts as a "constant", but it feels a bit weird to me somehow.

**Comment by XuehaiPan** [comment]
> I wouldn't want to allow float("inf"), as we currently don't allow any "call expressions" as default values

Fair enough. It can be simply replaced with `math.inf`.

> `math.inf` I could go either way on. `math.inf == math.inf` apparently, so I guess it counts as a "constant", but it feels a bit weird to me somehow.

It is a constant. One real-world application is calculating the `inf`-norm:

```python
def norm(vec: list[float], p: float = 2.0) -> float:
    if p == math.inf:
        return max(abs(x) for x in vec)
    if p == -math.inf:
        return min(abs(x) for x in vec)
    return pow(sum(pow(x, p) for x in vec), 1 / p)
```

**Comment by AlexWaygood** [comment]
Just to check: are you actually using these constants as defaults in real code, or is this just a theoretical concern?

If this is a real problem, I'm very happy to accept a PR allowing these attributes as defaults in stubs. But if it's just a theoretical concern, I'd probably prefer to leave the rules as they are.

We basically just have a hardcoded list of allowed attributes from the `sys` module at the moment, because we noticed it was reasonably common for `sys`-module constants to be used as default values. If this is a real problem, then I'm happy to add things to the hardcoded list of allowed attributes — but in an ideal world, I'd prefer for that list not to get too long, so I'd be inclined not to make the change if this is just a theoretical concern.

**Comment by XuehaiPan** [comment]
> Just to check: are you actually using these constants as default values in function signatures in real code, or is this just a theoretical concern?

I'm using `inf` and `-inf` for specifying an interval (defaults to the real number set `(-∞, +∞)`:

```python
class Box:
    def __init__(self, low=-inf, high=inf):
        ...

box = Box(low=0.0)  # represents [0, +∞)
```

However, I'm using `numpy.inf` rather than `math.inf` here.

I have read that [PEP 593](https://peps.python.org/pep-0593) is adding `typing.Annotated`:

```python
SmallNumber = Annotated[int, ValueRange(0, 10)]
```

but I haven't seen many details about `ValueRange`. Maybe it's stored as `str` or something. If it's a function or class, I think it could be:

```python
class ValueRange:
    def __init__(self, start, end = math.inf): ...
```

**Comment by AlexWaygood** [comment]
Cool — fine to add these if this stems from a real-world use case, then!

**Comment by JelleZijlstra** [comment]
`ValueRange` in PEP 593 is just an example of what third-party tools could do; it is not and is not intended to be defined as part of the core type system.
